### DISCUSSION TITLE
Rule not working for a soap+xml request having namespaces

### DISCUSSION THREAD

**Discussion by niharrbiswal** [mockoon/mockoon#754] [Help]
Here is the request XML and I'm trying to define a rule based on the PostCode .
Rule is defined as  : Body | Envelope.Body.fetchPolicies.request.PostCode | equals | 1 .  But that's not working.

```xml
<?xml version='1.0' encoding='UTF-8'?>
<soapenv:Envelope xmlns:soapenv="http://www.w3.org/2003/05/soap-envelope" xmlns:tns="http://services.global.com/us/portal/FetchService" xmlns:tns1="http://tns.mulesoft.com/json/1657025143167" xmlns:pogo="http://services.global.com/us/portal" xmlns:xsd="http://www.w3.org/2001/XMLSchema">
	<soapenv:Header/>
	<soapenv:Body>
		<tns:fetchPolicies>
			<tns:request>
				<pogo:PostCode>1</pogo:PostCode>
			</tns:request>
		</tns:fetchPolicies>
	</soapenv:Body>
</soapenv:Envelope>
```

Can anyone assist me get over it ?

**Comment by 255kb** [comment]
Mockoon's [XML support](https://mockoon.com/docs/latest/xml-support/) relies on a library translating XML to JSON.
It is a bit rough and provides the text value in a `_text` property.

Using this should do the trick:
`soapenv:Envelope.soapenv:Body.tns:fetchPolicies.tns:request.pogo:PostCode._text`

**Reply by niharrbiswal** [reply]
Thanks for your response. I already tried that and it's not working. Whether the selected content type of application/soap+xml has anything to do with it?

**Reply by 255kb** [reply]
Yes, we are only parsing XML when the content type is application/xml or text/xml. 
If you think application/soap+xml should be parsed too, feel free to open a new issue.

**Reply by niharrbiswal** [reply]
Thanks @255kb for your confirmation. I did open a new issue on this topic an hope to have this functionality sooner.

**Comment by mbijalwan1** [comment]
Can somebody pls help, how to define a rule for ChainCode="XD" from below xml. My Content-Type:Application.xml
AvailRequestSegment.HotelSearchCriteria.Criterion.HotelRef[2]._attributes.ChainCode seems not to be working when setting up Rules

```
<AvailRequestSegment InfoSource="MultiSource.AGGR">
	<HotelSearchCriteria AvailableOnlyIndicator="true">
		<Criterion ExactMatch="true" AlternateAvailability="Never">
			<HotelRef HotelCityCode="PAR"/>
			<HotelRef ChainCode="XD"/>``
```

**Reply by 255kb** [reply]
Using `AvailRequestSegment.HotelSearchCriteria.Criterion.HotelRef.2._attributes.ChainCode` should work. According to the [documentation](https://mockoon.com/docs/latest/templating/mockoon-request-helpers/#body), we use object-path which use `.number` to access array indexes.

**Reply by mbijalwan1** [reply]
Thanku Guillaume!!
 It worked now for me :-)
